src/source/CppGenerator.scala

```diff
@@ -210,7 +210,8 @@ class CppGenerator(spec: Spec) extends Generator(spec) {
 
     writeHppFile(cppName, origin, refs.hpp, refs.hppFwds, writeCppPrototype)
 
-    if (r.consts.nonEmpty || r.derivingTypes.nonEmpty) {
+    //If the only derivingType is AndroidParcelable, there is no need to generate the cpp file
+    if (r.consts.nonEmpty || (r.derivingTypes.nonEmpty && r.derivingTypes != Set(DerivingType.AndroidParcelable))) {
```

Comment:
It would be better if this explicitly checked for the derivings that it cares about, rather than only ruling out one of them.  If another new deriving is added in future, and it doesn't require editing this file to add new generated contents below, then it shouldn't require modifying this line either.